diff --git a/app2unit b/app2unit
index f7de122..18eadf1 100755
--- a/app2unit
+++ b/app2unit
@@ -1892,6 +1892,13 @@ gen_exec() {
 		read_entry_path "$A2U__ENTRY_PATH"
 	fi
 
+	service_apps="@SERVICE_APPS@"
+	for app in $service_apps; do
+		if [ "$A2U__ENTRY_ID" = "$app.desktop" ] || [ "$A2U__EXEC_NAME" = "$app" ]; then
+			A2U__UNIT_TYPE="service"
+		fi
+	done
+
 	# generate Unit ID as A2U__UNIT_ID
 	gen_unit_id
 
