diff --git a/app2unit b/app2unit
index 497b969..05fd67a 100755
--- a/app2unit
+++ b/app2unit
@@ -1455,6 +1455,13 @@ if [ -n "$ENTRY_PATH" ]; then
 	read_entry_path "$ENTRY_PATH" "$ENTRY_ACTION"
 fi
 
+service_apps="@SERVICE_APPS@"
+for app in $service_apps; do
+  if [ "$ENTRY_ID" = "$app.desktop" ] || [ "$EXEC_NAME" = "$app" ]; then
+    UNIT_TYPE="service"
+  fi
+done
+
 # generate Unit ID as UNIT_ID
 gen_unit_id
 
