diff --git a/configuration.go b/configuration.go
index 7b2ca16..ec7b673 100644
--- a/configuration.go
+++ b/configuration.go
@@ -179,6 +179,22 @@ func (config *Config) ReadConfig() error {
 		return fmt.Errorf("reading environment variables: %v", err)
 	}
 
+	if _, err := os.Stat("/etc/coordinates/latitude"); config.Lat != nil && err == nil {
+		if content, err := os.ReadFile("/etc/coordinates/latitude"); err == nil {
+			if lat, err := strconv.ParseFloat(string(content), 64); err == nil {
+				config.Lat = &lat
+			}
+		}
+	}
+
+	if _, err := os.Stat("/etc/coordinates/longitude"); config.Lng != nil && err == nil {
+		if content, err := os.ReadFile("/etc/coordinates/longitude"); err == nil {
+			if lng, err := strconv.ParseFloat(string(content), 64); err == nil {
+				config.Lng = &lng
+			}
+		}
+	}
+
 	slog.Info("Loaded configuration", "config", config)
 	return nil
 }
