diff --git a/service.go b/service.go
index 112ddb2..d58cf53 100644
--- a/service.go
+++ b/service.go
@@ -41,6 +41,24 @@ func NewService(initialMode Mode) Service {
 // AddListener adds a new callback to be run each time the current mode changes.
 func (service *Service) AddListener(listener ModeListener) {
 	*service.listeners = append(*service.listeners, listener)
+
+	// Apply once with the initial mode.
+	if err := listener.OnModeChanged(service.currentMode); err != nil {
+		fmt.Println("error applying initial mode:", err)
+	}
+}
+
+// Custom method for adding the script listener so we can skip the transition
+// if the initial mode is dark
+// (I made a new method for the sake of easy patching)
+func (service *Service) AddScriptListener(listener ModeListener) {
+	*service.listeners = append(*service.listeners, listener)
+
+	if (service.currentMode == "dark") {
+		log.Println("Skipping script listener initial OnModeChanged as initial mode is dark")
+		return
+	}
+
 	// Apply once with the initial mode.
 	if err := listener.OnModeChanged(service.currentMode); err != nil {
 		fmt.Println("error applying initial mode:", err)
@@ -222,7 +240,7 @@ func ExecuteService(ctx context.Context, config *Config, readyFd *os.File) error
 	log.Println("Initial mode set to:", initialMode)
 
 	service := NewService(initialMode)
-	service.AddListener(&ScriptRunner{})
+	service.AddScriptListener(&ScriptRunner{})
 	service.AddListener(CacheSaver{})
 
 	if err := setupListeners(&service, config, ctx, initialMode, initialLocation); err != nil {
