diff --git a/disko b/disko
index 807c4e1..285073d 100755
--- a/disko
+++ b/disko
@@ -92,6 +92,14 @@ while [[ $# -gt 0 ]]; do
     --show-trace)
       nix_args+=("$1")
       ;;
+    --override-input)
+      nix_args+=("$1" "$2" "$3")
+      shift
+      shift
+      ;;
+    --no-write-lock-file)
+      nix_args+=("$1")
+      ;;
     *)
       if [ -z ${disko_config+x} ]; then
         disko_config=$1
