diff --git a/homeassistant/components/husqvarna_automower/sensor.py b/homeassistant/components/husqvarna_automower/sensor.py
index 10aec9b153..479de4bff3 100644
--- a/homeassistant/components/husqvarna_automower/sensor.py
+++ b/homeassistant/components/husqvarna_automower/sensor.py
@@ -298,6 +298,7 @@ SENSOR_TYPES: tuple[AutomowerSensorEntityDescription, ...] = (
         key="next_start_timestamp",
         translation_key="next_start_timestamp",
         device_class=SensorDeviceClass.TIMESTAMP,
+        exists_fn=lambda data: data.planner.next_start_datetime is not None,
         value_fn=lambda data: dt_util.as_local(data.planner.next_start_datetime),
     ),
     AutomowerSensorEntityDescription(
