diff --git a/src/events/Windows.cpp b/src/events/Windows.cpp
index 7b0fbd34..398a8805 100644
--- a/src/events/Windows.cpp
+++ b/src/events/Windows.cpp
@@ -343,6 +343,7 @@ void Events::listener_mapWindow(void* owner, void* data) {
 
             PWINDOW->m_pWorkspace = pWorkspace;
             PWINDOW->m_pMonitor   = pWorkspace->m_monitor;
+            PMONITOR              = PWINDOW->m_pMonitor.lock();
 
             if (PWINDOW->m_pMonitor.lock()->activeSpecialWorkspace && !pWorkspace->m_isSpecialWorkspace)
                 workspaceSilent = true;
@@ -352,8 +353,6 @@ void Events::listener_mapWindow(void* owner, void* data) {
                     pWorkspace->m_monitor->setSpecialWorkspace(pWorkspace);
                 else if (PMONITOR->activeWorkspaceID() != REQUESTEDWORKSPACEID && !PWINDOW->m_bNoInitialFocus)
                     g_pKeybindManager->m_mDispatchers["workspace"](requestedWorkspaceName);
-
-                PMONITOR = g_pCompositor->m_lastMonitor.lock();
             }
 
             requestedFSMonitor = MONITOR_INVALID;
