diff --git a/src/events/Windows.cpp b/src/events/Windows.cpp
index 838839d3..68a545bc 100644
--- a/src/events/Windows.cpp
+++ b/src/events/Windows.cpp
@@ -355,6 +355,7 @@ void Events::listener_mapWindow(void* owner, void* data) {
 
             PWINDOW->m_workspace = pWorkspace;
             PWINDOW->m_monitor   = pWorkspace->m_monitor;
+            PMONITOR             = PWINDOW->m_monitor.lock();
 
             if (PWINDOW->m_monitor.lock()->m_activeSpecialWorkspace && !pWorkspace->m_isSpecialWorkspace)
                 workspaceSilent = true;
@@ -364,8 +365,6 @@ void Events::listener_mapWindow(void* owner, void* data) {
                     pWorkspace->m_monitor->setSpecialWorkspace(pWorkspace);
                 else if (PMONITOR->activeWorkspaceID() != REQUESTEDWORKSPACEID && !PWINDOW->m_noInitialFocus)
                     g_pKeybindManager->m_mDispatchers["workspace"](requestedWorkspaceName);
-
-                PMONITOR = g_pCompositor->m_lastMonitor.lock();
             }
 
             requestedFSMonitor = MONITOR_INVALID;
