diff --git a/src/events/Windows.cpp b/src/events/Windows.cpp
index f946ce03..2f7b8581 100644
--- a/src/events/Windows.cpp
+++ b/src/events/Windows.cpp
@@ -445,10 +445,12 @@ void Events::listener_mapWindow(void* owner, void* data) {
             } else if (r.szRule.starts_with("center")) {
                 auto       RESERVEDOFFSET = Vector2D();
                 const auto ARGS           = CVarList(r.szRule, 2, ' ');
-                if (ARGS[1] == "1")
-                    RESERVEDOFFSET = (PMONITOR->vecReservedTopLeft - PMONITOR->vecReservedBottomRight) / 2.f;
+                if (const auto PMONITOR = PWINDOW->m_pMonitor.lock(); PMONITOR) {
+                    if (ARGS[1] == "1")
+                        RESERVEDOFFSET = (PMONITOR->vecReservedTopLeft - PMONITOR->vecReservedBottomRight) / 2.f;
 
-                PWINDOW->m_vRealPosition = PMONITOR->middle() - PWINDOW->m_vRealSize.goal() / 2.f + RESERVEDOFFSET;
+                    PWINDOW->m_vRealPosition = PMONITOR->middle() - PWINDOW->m_vRealSize.goal() / 2.f + RESERVEDOFFSET;
+                }
             }
         }
 
