diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index c4d8734e..5584cd13 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -2093,13 +2093,13 @@ SDispatchResult CKeybindManager::toggleSpecialWorkspace(std::string args) {
         return {.success = false, .error = "Invalid workspace passed to special"};
     }
 
-    bool       requestedWorkspaceIsAlreadyOpen = false;
-    const auto PMONITOR                        = Desktop::focusState()->monitor();
-    auto       specialOpenOnMonitor            = PMONITOR->activeSpecialWorkspaceID();
+    bool requestedWorkspaceIsAlreadyOpen = false;
+    auto PMONITOR                        = Desktop::focusState()->monitor();
 
     for (auto const& m : g_pCompositor->m_monitors) {
         if (m->activeSpecialWorkspaceID() == workspaceID) {
             requestedWorkspaceIsAlreadyOpen = true;
+            PMONITOR                        = m;
             break;
         }
     }
@@ -2108,7 +2108,7 @@ SDispatchResult CKeybindManager::toggleSpecialWorkspace(std::string args) {
 
     PHLWORKSPACEREF focusedWorkspace;
 
-    if (requestedWorkspaceIsAlreadyOpen && specialOpenOnMonitor == workspaceID) {
+    if (requestedWorkspaceIsAlreadyOpen) {
         // already open on this monitor
         Log::logger->log(Log::DEBUG, "Toggling special workspace {} to closed", workspaceID);
         PMONITOR->setSpecialWorkspace(nullptr);
