diff --git a/src/Compositor.cpp b/src/Compositor.cpp
index 3d0b5173..6a2a9e49 100644
--- a/src/Compositor.cpp
+++ b/src/Compositor.cpp
@@ -915,9 +915,23 @@ PHLWINDOW CCompositor::vectorToWindowUnified(const Vector2D& pos, uint8_t proper
         }
     }
 
+    // prioritize always on top windows
+    std::vector<PHLWINDOW> alwaysOnTopWindows;
+    std::vector<PHLWINDOW> windows;
+    windows.reserve(g_pCompositor->m_windows.size());
+
+    for (auto const& w : m_windows) {
+        if (w->m_alwaysOnTop)
+            alwaysOnTopWindows.emplace_back(w);
+        else
+            windows.emplace_back(w);
+    }
+
+    windows.insert(windows.end(), alwaysOnTopWindows.begin(), alwaysOnTopWindows.end());
+
     auto windowForWorkspace = [&](bool special) -> PHLWINDOW {
         auto floating = [&](bool aboveFullscreen) -> PHLWINDOW {
-            for (auto const& w : m_windows | std::views::reverse) {
+            for (auto const& w : windows | std::views::reverse) {
 
                 if (special && !w->onSpecialWorkspace()) // because special floating may creep up into regular
                     continue;
@@ -2416,7 +2430,7 @@ void CCompositor::setWindowFullscreenState(const PHLWINDOW PWINDOW, SFullscreenS
 
     // make all windows on the same workspace under the fullscreen window
     for (auto const& w : m_windows) {
-        if (w->m_workspace == PWORKSPACE && !w->isFullscreen() && !w->m_fadingOut && !w->m_pinned)
+        if (w->m_workspace == PWORKSPACE && !w->isFullscreen() && !w->m_fadingOut && !w->m_pinned && !w->m_alwaysOnTop)
             w->m_createdOverFullscreen = false;
     }
 
diff --git a/src/debug/HyprCtl.cpp b/src/debug/HyprCtl.cpp
index a0ccddd0..2e6eaa1c 100644
--- a/src/debug/HyprCtl.cpp
+++ b/src/debug/HyprCtl.cpp
@@ -381,6 +381,7 @@ std::string CHyprCtl::getWindowData(PHLWINDOW w, eHyprCtlOutputFormat format) {
     "pid": {},
     "xwayland": {},
     "pinned": {},
+    "alwaysOnTop": {},
     "fullscreen": {},
     "fullscreenClient": {},
     "grouped": [{}],
@@ -395,22 +396,22 @@ std::string CHyprCtl::getWindowData(PHLWINDOW w, eHyprCtlOutputFormat format) {
             sc<int>(w->m_realPosition->goal().y), sc<int>(w->m_realSize->goal().x), sc<int>(w->m_realSize->goal().y), w->m_workspace ? w->workspaceID() : WORKSPACE_INVALID,
             escapeJSONStrings(!w->m_workspace ? "" : w->m_workspace->m_name), (sc<int>(w->m_isFloating) == 1 ? "true" : "false"), (w->m_isPseudotiled ? "true" : "false"),
             w->monitorID(), escapeJSONStrings(w->m_class), escapeJSONStrings(w->m_title), escapeJSONStrings(w->m_initialClass), escapeJSONStrings(w->m_initialTitle), w->getPID(),
-            (sc<int>(w->m_isX11) == 1 ? "true" : "false"), (w->m_pinned ? "true" : "false"), sc<uint8_t>(w->m_fullscreenState.internal), sc<uint8_t>(w->m_fullscreenState.client),
-            getGroupedData(w, format), getTagsData(w, format), rc<uintptr_t>(w->m_swallowed.get()), getFocusHistoryID(w),
+            (sc<int>(w->m_isX11) == 1 ? "true" : "false"), (w->m_pinned ? "true" : "false"), (w->m_alwaysOnTop ? "true" : "false"), sc<uint8_t>(w->m_fullscreenState.internal),
+            sc<uint8_t>(w->m_fullscreenState.client), getGroupedData(w, format), getTagsData(w, format), rc<uintptr_t>(w->m_swallowed.get()), getFocusHistoryID(w),
             (g_pInputManager->isWindowInhibiting(w, false) ? "true" : "false"), escapeJSONStrings(w->xdgTag().value_or("")), escapeJSONStrings(w->xdgDescription().value_or("")));
     } else {
         return std::format(
             "Window {:x} -> {}:\n\tmapped: {}\n\thidden: {}\n\tat: {},{}\n\tsize: {},{}\n\tworkspace: {} ({})\n\tfloating: {}\n\tpseudo: {}\n\tmonitor: {}\n\tclass: {}\n\ttitle: "
             "{}\n\tinitialClass: {}\n\tinitialTitle: {}\n\tpid: "
-            "{}\n\txwayland: {}\n\tpinned: "
+            "{}\n\txwayland: {}\n\tpinned: {}\n\talwaysOnTop: "
             "{}\n\tfullscreen: {}\n\tfullscreenClient: {}\n\tgrouped: {}\n\ttags: {}\n\tswallowing: {:x}\n\tfocusHistoryID: {}\n\tinhibitingIdle: {}\n\txdgTag: "
             "{}\n\txdgDescription: {}\n\n",
             rc<uintptr_t>(w.get()), w->m_title, sc<int>(w->m_isMapped), sc<int>(w->isHidden()), sc<int>(w->m_realPosition->goal().x), sc<int>(w->m_realPosition->goal().y),
             sc<int>(w->m_realSize->goal().x), sc<int>(w->m_realSize->goal().y), w->m_workspace ? w->workspaceID() : WORKSPACE_INVALID,
             (!w->m_workspace ? "" : w->m_workspace->m_name), sc<int>(w->m_isFloating), sc<int>(w->m_isPseudotiled), w->monitorID(), w->m_class, w->m_title, w->m_initialClass,
-            w->m_initialTitle, w->getPID(), sc<int>(w->m_isX11), sc<int>(w->m_pinned), sc<uint8_t>(w->m_fullscreenState.internal), sc<uint8_t>(w->m_fullscreenState.client),
-            getGroupedData(w, format), getTagsData(w, format), rc<uintptr_t>(w->m_swallowed.get()), getFocusHistoryID(w), sc<int>(g_pInputManager->isWindowInhibiting(w, false)),
-            w->xdgTag().value_or(""), w->xdgDescription().value_or(""));
+            w->m_initialTitle, w->getPID(), sc<int>(w->m_isX11), sc<int>(w->m_pinned), (int)w->m_alwaysOnTop, sc<uint8_t>(w->m_fullscreenState.internal),
+            sc<uint8_t>(w->m_fullscreenState.client), getGroupedData(w, format), getTagsData(w, format), rc<uintptr_t>(w->m_swallowed.get()), getFocusHistoryID(w),
+            sc<int>(g_pInputManager->isWindowInhibiting(w, false)), w->xdgTag().value_or(""), w->xdgDescription().value_or(""));
     }
 }
 
diff --git a/src/desktop/Window.hpp b/src/desktop/Window.hpp
index 0a7e207c..72b0d498 100644
--- a/src/desktop/Window.hpp
+++ b/src/desktop/Window.hpp
@@ -241,6 +241,9 @@ class CWindow {
     // For pinned (sticky) windows
     bool m_pinned = false;
 
+    // For placing above other windows in the workspace
+    bool m_alwaysOnTop = false;
+
     // For preserving pinned state when fullscreening a pinned window
     bool m_pinFullscreened = false;
 
diff --git a/src/desktop/WindowRule.cpp b/src/desktop/WindowRule.cpp
index dc6564ca..b264b439 100644
--- a/src/desktop/WindowRule.cpp
+++ b/src/desktop/WindowRule.cpp
@@ -5,7 +5,7 @@
 #include "../config/ConfigManager.hpp"
 
 static const auto RULES = std::unordered_set<std::string>{
-    "float", "fullscreen", "maximize", "noinitialfocus", "pin", "stayfocused", "tile", "renderunfocused", "persistentsize",
+    "float", "fullscreen", "maximize", "noinitialfocus", "pin", "alwaysontop", "stayfocused", "tile", "renderunfocused", "persistentsize",
 };
 static const auto RULES_PREFIX = std::unordered_set<std::string>{
     "animation",     "bordercolor", "bordersize", "center",  "content", "fullscreenstate", "group",    "idleinhibit",   "maxsize",     "minsize",        "monitor",
@@ -32,6 +32,8 @@ CWindowRule::CWindowRule(const std::string& rule, const std::string& value, bool
         m_ruleType = RULE_NOINITIALFOCUS;
     else if (rule == "pin")
         m_ruleType = RULE_PIN;
+    else if (rule == "alwaysontop")
+        m_ruleType = RULE_ALWAYSONTOP;
     else if (rule == "stayfocused")
         m_ruleType = RULE_STAYFOCUSED;
     else if (rule == "tile")
diff --git a/src/desktop/WindowRule.hpp b/src/desktop/WindowRule.hpp
index 5bf462e9..6b133e77 100644
--- a/src/desktop/WindowRule.hpp
+++ b/src/desktop/WindowRule.hpp
@@ -15,6 +15,7 @@ class CWindowRule {
         RULE_MAXIMIZE,
         RULE_NOINITIALFOCUS,
         RULE_PIN,
+        RULE_ALWAYSONTOP,
         RULE_STAYFOCUSED,
         RULE_TILE,
         RULE_RENDERUNFOCUSED,
diff --git a/src/events/Windows.cpp b/src/events/Windows.cpp
index 16f71534..ab448fb6 100644
--- a/src/events/Windows.cpp
+++ b/src/events/Windows.cpp
@@ -243,6 +243,10 @@ void Events::listener_mapWindow(void* owner, void* data) {
                 PWINDOW->m_pinned = true;
                 break;
             }
+            case CWindowRule::RULE_ALWAYSONTOP: {
+                PWINDOW->m_alwaysOnTop = true;
+                break;
+            }
             case CWindowRule::RULE_FULLSCREEN: {
                 requestedInternalFSMode = FSMODE_FULLSCREEN;
                 break;
@@ -333,6 +337,9 @@ void Events::listener_mapWindow(void* owner, void* data) {
     if (PWINDOW->m_pinned && !PWINDOW->m_isFloating)
         PWINDOW->m_pinned = false;
 
+    if (PWINDOW->m_alwaysOnTop && !PWINDOW->m_isFloating)
+        PWINDOW->m_alwaysOnTop = false;
+
     CVarList WORKSPACEARGS = CVarList(requestedWorkspace, 0, ' ');
 
     if (!WORKSPACEARGS[0].empty()) {
diff --git a/src/layout/IHyprLayout.cpp b/src/layout/IHyprLayout.cpp
index 38868424..c5a2a40a 100644
--- a/src/layout/IHyprLayout.cpp
+++ b/src/layout/IHyprLayout.cpp
@@ -739,7 +739,8 @@ void IHyprLayout::changeWindowFloatingMode(PHLWINDOW pWindow) {
         g_pCompositor->setWindowFullscreenInternal(pWindow, FSMODE_NONE);
     }
 
-    pWindow->m_pinned = false;
+    pWindow->m_pinned      = false;
+    pWindow->m_alwaysOnTop = false;
 
     g_pHyprRenderer->damageWindow(pWindow, true);
 
diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index 68750ad7..dae3cd53 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -129,6 +129,7 @@ CKeybindManager::CKeybindManager() {
     m_dispatchers["swapnext"]                       = swapnext;
     m_dispatchers["swapactiveworkspaces"]           = swapActiveWorkspaces;
     m_dispatchers["pin"]                            = pinActive;
+    m_dispatchers["togglealwaysontop"]              = toggleActiveAlwaysOnTop;
     m_dispatchers["mouse"]                          = mouse;
     m_dispatchers["bringactivetotop"]               = bringActiveToTop;
     m_dispatchers["alterzorder"]                    = alterZOrder;
@@ -2769,6 +2770,31 @@ SDispatchResult CKeybindManager::pinActive(std::string args) {
     return {};
 }
 
+SDispatchResult CKeybindManager::toggleActiveAlwaysOnTop(std::string args) {
+
+    PHLWINDOW PWINDOW = nullptr;
+
+    if (args != "active" && args.length() > 1)
+        PWINDOW = g_pCompositor->getWindowByRegex(args);
+    else
+        PWINDOW = g_pCompositor->m_lastWindow.lock();
+
+    if (!PWINDOW) {
+        Debug::log(ERR, "togglealwaysontop: window not found");
+        return {.success = false, .error = "togglealwaysontop: window not found"};
+    }
+
+    if (!PWINDOW->m_isFloating || PWINDOW->isFullscreen())
+        return {};
+
+    PWINDOW->m_alwaysOnTop = !PWINDOW->m_alwaysOnTop;
+
+    PWINDOW->updateDynamicRules();
+    g_pCompositor->updateWindowAnimatedDecorationValues(PWINDOW);
+
+    return {};
+}
+
 SDispatchResult CKeybindManager::mouse(std::string args) {
     const auto ARGS    = CVarList(args.substr(1), 2, ' ');
     const auto PRESSED = args[0] == '1';
diff --git a/src/managers/KeybindManager.hpp b/src/managers/KeybindManager.hpp
index 592588b5..e233d1c8 100644
--- a/src/managers/KeybindManager.hpp
+++ b/src/managers/KeybindManager.hpp
@@ -223,6 +223,7 @@ class CKeybindManager {
     static SDispatchResult swapnext(std::string);
     static SDispatchResult swapActiveWorkspaces(std::string);
     static SDispatchResult pinActive(std::string);
+    static SDispatchResult toggleActiveAlwaysOnTop(std::string);
     static SDispatchResult mouse(std::string);
     static SDispatchResult bringActiveToTop(std::string);
     static SDispatchResult alterZOrder(std::string);
diff --git a/src/render/Renderer.cpp b/src/render/Renderer.cpp
index 14ed1469..0f1b80eb 100644
--- a/src/render/Renderer.cpp
+++ b/src/render/Renderer.cpp
@@ -351,8 +351,8 @@ void CHyprRenderer::renderWorkspaceWindowsFullscreen(PHLMONITOR pMonitor, PHLWOR
 
     // then render windows over fullscreen.
     for (auto const& w : g_pCompositor->m_windows) {
-        if (w->workspaceID() != pWorkspaceWindow->workspaceID() || !w->m_isFloating || (!w->m_createdOverFullscreen && !w->m_pinned) || (!w->m_isMapped && !w->m_fadingOut) ||
-            w->isFullscreen())
+        if (w->workspaceID() != pWorkspaceWindow->workspaceID() || !w->m_isFloating || (!w->m_createdOverFullscreen && !w->m_pinned && !w->m_alwaysOnTop) ||
+            (!w->m_isMapped && !w->m_fadingOut) || w->isFullscreen())
             continue;
 
         if (w->m_monitor == pWorkspace->m_monitor && pWorkspace->m_isSpecialWorkspace != w->onSpecialWorkspace())
@@ -446,7 +446,28 @@ void CHyprRenderer::renderWorkspaceWindows(PHLMONITOR pMonitor, PHLWORKSPACE pWo
         if (!w)
             continue;
 
-        if (!w->m_isFloating || w->m_pinned)
+        if (!w->m_isFloating || w->m_pinned || w->m_alwaysOnTop)
+            continue;
+
+        // some things may force us to ignore the special/not special disparity
+        const bool IGNORE_SPECIAL_CHECK = w->m_monitorMovedFrom != -1 && (w->m_workspace && !w->m_workspace->isVisible());
+
+        if (!IGNORE_SPECIAL_CHECK && pWorkspace->m_isSpecialWorkspace != w->onSpecialWorkspace())
+            continue;
+
+        if (pWorkspace->m_isSpecialWorkspace && w->m_monitor != pWorkspace->m_monitor)
+            continue; // special on another are rendered as a part of the base pass
+
+        // render the bad boy
+        renderWindow(w.lock(), pMonitor, time, true, RENDER_PASS_ALL);
+    }
+
+    // always on top floating windows
+    for (auto& w : windows) {
+        if (!w)
+            continue;
+
+        if (!w->m_alwaysOnTop || !w->m_isFloating || w->m_pinned)
             continue;
 
         // some things may force us to ignore the special/not special disparity
