diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index 4423bbdb..fbfd5756 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -2104,13 +2104,13 @@ SDispatchResult CKeybindManager::toggleSpecialWorkspace(std::string args) {
         return {.success = false, .error = "Invalid workspace passed to special"};
     }
 
-    bool       requestedWorkspaceIsAlreadyOpen = false;
-    const auto PMONITOR                        = Desktop::focusState()->monitor();
-    auto       specialOpenOnMonitor            = PMONITOR->activeSpecialWorkspaceID();
+    bool requestedWorkspaceIsAlreadyOpen = false;
+    auto PMONITOR                        = Desktop::focusState()->monitor();
 
     for (auto const& m : g_pCompositor->m_monitors) {
         if (m->activeSpecialWorkspaceID() == workspaceID) {
             requestedWorkspaceIsAlreadyOpen = true;
+            PMONITOR                        = m;
             break;
         }
     }
@@ -2119,8 +2119,7 @@ SDispatchResult CKeybindManager::toggleSpecialWorkspace(std::string args) {
 
     PHLWORKSPACEREF focusedWorkspace;
 
-    if (requestedWorkspaceIsAlreadyOpen && specialOpenOnMonitor == workspaceID) {
-        // already open on this monitor
+    if (requestedWorkspaceIsAlreadyOpen) {
         Debug::log(LOG, "Toggling special workspace {} to closed", workspaceID);
         PMONITOR->setSpecialWorkspace(nullptr);
 
