diff --git a/src/config/shared/workspace/WorkspaceRule.cpp b/src/config/shared/workspace/WorkspaceRule.cpp
index 28d5fcab..73755414 100644
--- a/src/config/shared/workspace/WorkspaceRule.cpp
+++ b/src/config/shared/workspace/WorkspaceRule.cpp
@@ -3,7 +3,7 @@
 using namespace Config;
 
 void CWorkspaceRule::mergeLeft(const CWorkspaceRule& other) {
-    if (m_monitor.empty())
+    if (!other.m_monitor.empty())
         m_monitor = other.m_monitor;
     if (m_workspaceString.empty())
         m_workspaceString = other.m_workspaceString;
@@ -14,8 +14,7 @@ void CWorkspaceRule::mergeLeft(const CWorkspaceRule& other) {
 
     if (other.m_isDefault)
         m_isDefault = true;
-    if (other.m_isPersistent)
-        m_isPersistent = true;
+    m_isPersistent = other.m_isPersistent;
     if (other.m_gapsIn.has_value())
         m_gapsIn = other.m_gapsIn;
     if (other.m_gapsOut.has_value())
