diff --git a/src/config/ConfigManager.cpp b/src/config/ConfigManager.cpp
index cd5b0ec8..7a89f681 100644
--- a/src/config/ConfigManager.cpp
+++ b/src/config/ConfigManager.cpp
@@ -2944,7 +2944,8 @@ std::optional<std::string> CConfigManager::handleGesture(const std::string& comm
     else if (data[startDataIdx] == "workspace")
         result = g_pTrackpadGestures->addGesture(makeUnique<CWorkspaceSwipeGesture>(), fingerCount, direction, modMask, deltaScale, disableInhibit);
     else if (data[startDataIdx] == "resize")
-        result = g_pTrackpadGestures->addGesture(makeUnique<CResizeTrackpadGesture>(), fingerCount, direction, modMask, deltaScale, disableInhibit);
+        result =
+            g_pTrackpadGestures->addGesture(makeUnique<CResizeTrackpadGesture>(std::string{data[startDataIdx + 2]}), fingerCount, direction, modMask, deltaScale, disableInhibit);
     else if (data[startDataIdx] == "move")
         result = g_pTrackpadGestures->addGesture(makeUnique<CMoveTrackpadGesture>(), fingerCount, direction, modMask, deltaScale, disableInhibit);
     else if (data[startDataIdx] == "special")
diff --git a/src/managers/input/trackpad/gestures/ResizeGesture.cpp b/src/managers/input/trackpad/gestures/ResizeGesture.cpp
index ffc7704b..cdd71412 100644
--- a/src/managers/input/trackpad/gestures/ResizeGesture.cpp
+++ b/src/managers/input/trackpad/gestures/ResizeGesture.cpp
@@ -5,6 +5,18 @@
 #include "../../../../render/Renderer.hpp"
 #include "../../../../layout/LayoutManager.hpp"
 
+CResizeTrackpadGesture::CResizeTrackpadGesture(const std::string_view& mode) {
+    std::string lc = std::string{mode};
+    std::ranges::transform(lc, lc.begin(), ::tolower);
+
+    if (lc.starts_with("normal"))
+        m_mode = MODE_NORMAL;
+    else if (lc.starts_with("dynamic"))
+        m_mode = MODE_DYNAMIC;
+    else
+        m_mode = MODE_NORMAL;
+}
+
 void CResizeTrackpadGesture::begin(const ITrackpadGesture::STrackpadGestureBegin& e) {
     ITrackpadGesture::begin(e);
 
@@ -15,6 +27,15 @@ void CResizeTrackpadGesture::update(const ITrackpadGesture::STrackpadGestureUpda
     if (!m_window)
         return;
 
+    const auto DELTA = e.swipe ? e.swipe->delta : e.pinch->delta;
+
+    if (m_mode == MODE_DYNAMIC && m_window->m_isFloating) {
+        g_layoutManager->moveTarget(e.swipe ? e.swipe->delta : e.pinch->delta, m_window->layoutTarget());
+        m_window->m_realSize->warp();
+        m_window->m_realPosition->warp();
+        return;
+    }
+
     g_pHyprRenderer->damageWindow(m_window.lock());
 
     g_layoutManager->resizeTarget((e.swipe ? e.swipe->delta : e.pinch->delta), m_window->layoutTarget(),
diff --git a/src/managers/input/trackpad/gestures/ResizeGesture.hpp b/src/managers/input/trackpad/gestures/ResizeGesture.hpp
index 0b47a224..bd30f5b6 100644
--- a/src/managers/input/trackpad/gestures/ResizeGesture.hpp
+++ b/src/managers/input/trackpad/gestures/ResizeGesture.hpp
@@ -6,7 +6,7 @@
 
 class CResizeTrackpadGesture : public ITrackpadGesture {
   public:
-    CResizeTrackpadGesture()          = default;
+    CResizeTrackpadGesture(const std::string_view& mode);
     virtual ~CResizeTrackpadGesture() = default;
 
     virtual void begin(const ITrackpadGesture::STrackpadGestureBegin& e);
@@ -15,4 +15,11 @@ class CResizeTrackpadGesture : public ITrackpadGesture {
 
   private:
     PHLWINDOWREF m_window;
+
+    enum eMode : uint8_t {
+        MODE_NORMAL = 0,
+        MODE_DYNAMIC,
+    };
+
+    eMode m_mode = MODE_NORMAL;
 };
