diff --git a/src/Compositor.cpp b/src/Compositor.cpp
index 154ec135..fd393cd0 100644
--- a/src/Compositor.cpp
+++ b/src/Compositor.cpp
@@ -888,9 +888,23 @@ PHLWINDOW CCompositor::vectorToWindowUnified(const Vector2D& pos, uint8_t proper
         }
     }
 
+    // prioritize always on top windows
+    std::vector<PHLWINDOWREF> alwaysOnTopWindows;
+    std::vector<PHLWINDOWREF> windows;
+    windows.reserve(g_pCompositor->m_windows.size());
+
+    for (auto const& w : m_windows) {
+        if (w->m_alwaysOnTop)
+            alwaysOnTopWindows.emplace_back(w);
+        else
+            windows.emplace_back(w);
+    }
+
+    windows.insert(windows.end(), alwaysOnTopWindows.begin(), alwaysOnTopWindows.end());
+
     auto windowForWorkspace = [&](bool special) -> PHLWINDOW {
         auto floating = [&](bool aboveFullscreen) -> PHLWINDOW {
-            for (auto const& w : m_windows | std::views::reverse) {
+            for (auto const& w : windows | std::views::reverse) {
 
                 if (special && !w->onSpecialWorkspace()) // because special floating may creep up into regular
                     continue;
@@ -924,12 +938,12 @@ PHLWINDOW CCompositor::vectorToWindowUnified(const Vector2D& pos, uint8_t proper
                             // TODO: this is wrong, we should focus the parent, but idk how to get it considering it's nullptr in most cases.
                         }
 
-                        return w;
+                        return w.lock();
                     }
 
                     if (!w->m_isX11) {
                         if (w->hasPopupAt(pos))
-                            return w;
+                            return w.lock();
                     }
                 }
             }
@@ -2362,7 +2376,7 @@ void CCompositor::setWindowFullscreenState(const PHLWINDOW PWINDOW, SFullscreenS
 
     // make all windows on the same workspace under the fullscreen window
     for (auto const& w : m_windows) {
-        if (w->m_workspace == PWORKSPACE && !w->isFullscreen() && !w->m_fadingOut && !w->m_pinned)
+        if (w->m_workspace == PWORKSPACE && !w->isFullscreen() && !w->m_fadingOut && !w->m_pinned && !w->m_alwaysOnTop)
             w->m_createdOverFullscreen = false;
     }
 
diff --git a/src/debug/HyprCtl.cpp b/src/debug/HyprCtl.cpp
index 2ad182ab..a323d82b 100644
--- a/src/debug/HyprCtl.cpp
+++ b/src/debug/HyprCtl.cpp
@@ -277,6 +277,7 @@ std::string CHyprCtl::getWindowData(PHLWINDOW w, eHyprCtlOutputFormat format) {
     "pid": {},
     "xwayland": {},
     "pinned": {},
+    "alwaysOnTop": {},
     "fullscreen": {},
     "fullscreenClient": {},
     "grouped": [{}],
@@ -291,21 +292,22 @@ std::string CHyprCtl::getWindowData(PHLWINDOW w, eHyprCtlOutputFormat format) {
             (int)w->m_realSize->goal().x, (int)w->m_realSize->goal().y, w->m_workspace ? w->workspaceID() : WORKSPACE_INVALID,
             escapeJSONStrings(!w->m_workspace ? "" : w->m_workspace->m_name), ((int)w->m_isFloating == 1 ? "true" : "false"), (w->m_isPseudotiled ? "true" : "false"),
             (int64_t)w->monitorID(), escapeJSONStrings(w->m_class), escapeJSONStrings(w->m_title), escapeJSONStrings(w->m_initialClass), escapeJSONStrings(w->m_initialTitle),
-            w->getPID(), ((int)w->m_isX11 == 1 ? "true" : "false"), (w->m_pinned ? "true" : "false"), (uint8_t)w->m_fullscreenState.internal, (uint8_t)w->m_fullscreenState.client,
-            getGroupedData(w, format), getTagsData(w, format), (uintptr_t)w->m_swallowed.get(), getFocusHistoryID(w),
+            w->getPID(), ((int)w->m_isX11 == 1 ? "true" : "false"), (w->m_pinned ? "true" : "false"), (w->m_alwaysOnTop ? "true" : "false"), (uint8_t)w->m_fullscreenState.internal,
+            (uint8_t)w->m_fullscreenState.client, getGroupedData(w, format), getTagsData(w, format), (uintptr_t)w->m_swallowed.get(), getFocusHistoryID(w),
             (g_pInputManager->isWindowInhibiting(w, false) ? "true" : "false"), escapeJSONStrings(w->xdgTag().value_or("")), escapeJSONStrings(w->xdgDescription().value_or("")));
     } else {
         return std::format(
             "Window {:x} -> {}:\n\tmapped: {}\n\thidden: {}\n\tat: {},{}\n\tsize: {},{}\n\tworkspace: {} ({})\n\tfloating: {}\n\tpseudo: {}\n\tmonitor: {}\n\tclass: {}\n\ttitle: "
             "{}\n\tinitialClass: {}\n\tinitialTitle: {}\n\tpid: "
-            "{}\n\txwayland: {}\n\tpinned: "
+            "{}\n\txwayland: {}\n\tpinned: {}\n\talwaysOnTop: "
             "{}\n\tfullscreen: {}\n\tfullscreenClient: {}\n\tgrouped: {}\n\ttags: {}\n\tswallowing: {:x}\n\tfocusHistoryID: {}\n\tinhibitingIdle: {}\n\txdgTag: "
             "{}\n\txdgDescription: {}\n\n",
             (uintptr_t)w.get(), w->m_title, (int)w->m_isMapped, (int)w->isHidden(), (int)w->m_realPosition->goal().x, (int)w->m_realPosition->goal().y,
             (int)w->m_realSize->goal().x, (int)w->m_realSize->goal().y, w->m_workspace ? w->workspaceID() : WORKSPACE_INVALID, (!w->m_workspace ? "" : w->m_workspace->m_name),
             (int)w->m_isFloating, (int)w->m_isPseudotiled, (int64_t)w->monitorID(), w->m_class, w->m_title, w->m_initialClass, w->m_initialTitle, w->getPID(), (int)w->m_isX11,
-            (int)w->m_pinned, (uint8_t)w->m_fullscreenState.internal, (uint8_t)w->m_fullscreenState.client, getGroupedData(w, format), getTagsData(w, format),
-            (uintptr_t)w->m_swallowed.get(), getFocusHistoryID(w), (int)g_pInputManager->isWindowInhibiting(w, false), w->xdgTag().value_or(""), w->xdgDescription().value_or(""));
+            (int)w->m_pinned, (int)w->m_alwaysOnTop, (uint8_t)w->m_fullscreenState.internal, (uint8_t)w->m_fullscreenState.client, getGroupedData(w, format),
+            getTagsData(w, format), (uintptr_t)w->m_swallowed.get(), getFocusHistoryID(w), (int)g_pInputManager->isWindowInhibiting(w, false), w->xdgTag().value_or(""),
+            w->xdgDescription().value_or(""));
     }
 }
 
diff --git a/src/desktop/Window.hpp b/src/desktop/Window.hpp
index cc62dd3b..c16ef6ad 100644
--- a/src/desktop/Window.hpp
+++ b/src/desktop/Window.hpp
@@ -240,6 +240,9 @@ class CWindow {
     // For pinned (sticky) windows
     bool m_pinned = false;
 
+    // For placing above other windows in the workspace
+    bool m_alwaysOnTop = false;
+
     // For preserving pinned state when fullscreening a pinned window
     bool m_pinFullscreened = false;
 
diff --git a/src/desktop/WindowRule.cpp b/src/desktop/WindowRule.cpp
index db1b0dde..69f7f028 100644
--- a/src/desktop/WindowRule.cpp
+++ b/src/desktop/WindowRule.cpp
@@ -5,7 +5,7 @@
 #include "../config/ConfigManager.hpp"
 
 static const auto RULES = std::unordered_set<std::string>{
-    "float", "fullscreen", "maximize", "noinitialfocus", "pin", "stayfocused", "tile", "renderunfocused", "persistentsize",
+    "float", "fullscreen", "maximize", "noinitialfocus", "pin", "alwaysontop", "stayfocused", "tile", "renderunfocused", "persistentsize",
 };
 static const auto RULES_PREFIX = std::unordered_set<std::string>{
     "animation",     "bordercolor", "bordersize", "center",  "content", "fullscreenstate", "group",    "idleinhibit",   "maxsize",     "minsize",        "monitor",
@@ -33,6 +33,8 @@ CWindowRule::CWindowRule(const std::string& rule, const std::string& value, bool
         m_ruleType = RULE_NOINITIALFOCUS;
     else if (rule == "pin")
         m_ruleType = RULE_PIN;
+    else if (rule == "alwaysontop")
+        m_ruleType = RULE_ALWAYSONTOP;
     else if (rule == "stayfocused")
         m_ruleType = RULE_STAYFOCUSED;
     else if (rule == "tile")
diff --git a/src/desktop/WindowRule.hpp b/src/desktop/WindowRule.hpp
index 9af3909a..0a3ea767 100644
--- a/src/desktop/WindowRule.hpp
+++ b/src/desktop/WindowRule.hpp
@@ -15,6 +15,7 @@ class CWindowRule {
         RULE_MAXIMIZE,
         RULE_NOINITIALFOCUS,
         RULE_PIN,
+        RULE_ALWAYSONTOP,
         RULE_STAYFOCUSED,
         RULE_TILE,
         RULE_RENDERUNFOCUSED,
diff --git a/src/events/Windows.cpp b/src/events/Windows.cpp
index 838839d3..2ef25806 100644
--- a/src/events/Windows.cpp
+++ b/src/events/Windows.cpp
@@ -248,6 +248,10 @@ void Events::listener_mapWindow(void* owner, void* data) {
                 PWINDOW->m_pinned = true;
                 break;
             }
+            case CWindowRule::RULE_ALWAYSONTOP: {
+                PWINDOW->m_alwaysOnTop = true;
+                break;
+            }
             case CWindowRule::RULE_FULLSCREEN: {
                 requestedInternalFSMode = FSMODE_FULLSCREEN;
                 break;
@@ -337,6 +341,9 @@ void Events::listener_mapWindow(void* owner, void* data) {
     if (PWINDOW->m_pinned && !PWINDOW->m_isFloating)
         PWINDOW->m_pinned = false;
 
+    if (PWINDOW->m_alwaysOnTop && !PWINDOW->m_isFloating)
+        PWINDOW->m_alwaysOnTop = false;
+
     const CVarList WORKSPACEARGS = CVarList(requestedWorkspace, 0, ' ');
 
     if (!WORKSPACEARGS[0].empty()) {
diff --git a/src/layout/IHyprLayout.cpp b/src/layout/IHyprLayout.cpp
index 51ea5883..c073a89c 100644
--- a/src/layout/IHyprLayout.cpp
+++ b/src/layout/IHyprLayout.cpp
@@ -704,7 +704,8 @@ void IHyprLayout::changeWindowFloatingMode(PHLWINDOW pWindow) {
         g_pCompositor->setWindowFullscreenInternal(pWindow, FSMODE_NONE);
     }
 
-    pWindow->m_pinned = false;
+    pWindow->m_pinned      = false;
+    pWindow->m_alwaysOnTop = false;
 
     g_pHyprRenderer->damageWindow(pWindow, true);
 
diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index 1bd7e1a2..a96244e7 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -126,6 +126,7 @@ CKeybindManager::CKeybindManager() {
     m_mDispatchers["swapnext"]                       = swapnext;
     m_mDispatchers["swapactiveworkspaces"]           = swapActiveWorkspaces;
     m_mDispatchers["pin"]                            = pinActive;
+    m_mDispatchers["togglealwaysontop"]              = toggleActiveAlwaysOnTop;
     m_mDispatchers["mouse"]                          = mouse;
     m_mDispatchers["bringactivetotop"]               = bringActiveToTop;
     m_mDispatchers["alterzorder"]                    = alterZOrder;
@@ -2789,6 +2790,31 @@ SDispatchResult CKeybindManager::pinActive(std::string args) {
     return {};
 }
 
+SDispatchResult CKeybindManager::toggleActiveAlwaysOnTop(std::string args) {
+
+    PHLWINDOW PWINDOW = nullptr;
+
+    if (args != "active" && args.length() > 1)
+        PWINDOW = g_pCompositor->getWindowByRegex(args);
+    else
+        PWINDOW = g_pCompositor->m_lastWindow.lock();
+
+    if (!PWINDOW) {
+        Debug::log(ERR, "togglealwaysontop: window not found");
+        return {.success = false, .error = "togglealwaysontop: window not found"};
+    }
+
+    if (!PWINDOW->m_isFloating || PWINDOW->isFullscreen())
+        return {};
+
+    PWINDOW->m_alwaysOnTop = !PWINDOW->m_alwaysOnTop;
+
+    PWINDOW->updateDynamicRules();
+    g_pCompositor->updateWindowAnimatedDecorationValues(PWINDOW);
+
+    return {};
+}
+
 SDispatchResult CKeybindManager::mouse(std::string args) {
     const auto ARGS    = CVarList(args.substr(1), 2, ' ');
     const auto PRESSED = args[0] == '1';
diff --git a/src/managers/KeybindManager.hpp b/src/managers/KeybindManager.hpp
index ede16402..67766a3d 100644
--- a/src/managers/KeybindManager.hpp
+++ b/src/managers/KeybindManager.hpp
@@ -212,6 +212,7 @@ class CKeybindManager {
     static SDispatchResult swapnext(std::string);
     static SDispatchResult swapActiveWorkspaces(std::string);
     static SDispatchResult pinActive(std::string);
+    static SDispatchResult toggleActiveAlwaysOnTop(std::string);
     static SDispatchResult mouse(std::string);
     static SDispatchResult bringActiveToTop(std::string);
     static SDispatchResult alterZOrder(std::string);
diff --git a/src/render/Renderer.cpp b/src/render/Renderer.cpp
index 445c39cd..0c333f1b 100644
--- a/src/render/Renderer.cpp
+++ b/src/render/Renderer.cpp
@@ -338,8 +338,8 @@ void CHyprRenderer::renderWorkspaceWindowsFullscreen(PHLMONITOR pMonitor, PHLWOR
 
     // then render windows over fullscreen.
     for (auto const& w : g_pCompositor->m_windows) {
-        if (w->m_workspace != pWorkspaceWindow->m_workspace || !w->m_isFloating || (!w->m_createdOverFullscreen && !w->m_pinned) || (!w->m_isMapped && !w->m_fadingOut) ||
-            w->isFullscreen())
+        if (w->m_workspace != pWorkspaceWindow->m_workspace || !w->m_isFloating || (!w->m_createdOverFullscreen && !w->m_pinned && !w->m_alwaysOnTop) ||
+            (!w->m_isMapped && !w->m_fadingOut) || w->isFullscreen())
             continue;
 
         if (w->m_monitor == pWorkspace->m_monitor && pWorkspace->m_isSpecialWorkspace != w->onSpecialWorkspace())
@@ -433,6 +433,33 @@ void CHyprRenderer::renderWorkspaceWindows(PHLMONITOR pMonitor, PHLWORKSPACE pWo
         if (!w)
             continue;
 
+        if (w->m_alwaysOnTop)
+            continue;
+
+        if (!w->m_isFloating || w->m_pinned)
+            continue;
+
+        // some things may force us to ignore the special/not special disparity
+        const bool IGNORE_SPECIAL_CHECK = w->m_monitorMovedFrom != -1 && (w->m_workspace && !w->m_workspace->isVisible());
+
+        if (!IGNORE_SPECIAL_CHECK && pWorkspace->m_isSpecialWorkspace != w->onSpecialWorkspace())
+            continue;
+
+        if (pWorkspace->m_isSpecialWorkspace && w->m_monitor != pWorkspace->m_monitor)
+            continue; // special on another are rendered as a part of the base pass
+
+        // render the bad boy
+        renderWindow(w.lock(), pMonitor, time, true, RENDER_PASS_ALL);
+    }
+
+    // always on top floating windows
+    for (auto& w : windows) {
+        if (!w)
+            continue;
+
+        if (!w->m_alwaysOnTop)
+            continue;
+
         if (!w->m_isFloating || w->m_pinned)
             continue;
 
