diff --git a/src/Compositor.cpp b/src/Compositor.cpp
index 844a19f9..9cd94af7 100644
--- a/src/Compositor.cpp
+++ b/src/Compositor.cpp
@@ -2489,7 +2489,7 @@ Vector2D CCompositor::parseWindowVectorArgsRelative(const std::string& args, con
         y          = y.substr(0, y.length() - 1);
     }
 
-    if (!isNumber2(x) || !isNumber2(y)) {
+    if (!isNumber2(x, xIsPercent) || !isNumber2(y, yIsPercent)) {
         Log::logger->log(Log::ERR, "parseWindowVectorArgsRelative: args not numbers");
         return relativeTo;
     }
