diff --git a/src/Compositor.cpp b/src/Compositor.cpp
index 1daccbfd..c85ddc8c 100644
--- a/src/Compositor.cpp
+++ b/src/Compositor.cpp
@@ -873,9 +873,23 @@ PHLWINDOW CCompositor::vectorToWindowUnified(const Vector2D& pos, uint8_t proper
         }
     }
 
+    // prioritize always on top windows
+    std::vector<PHLWINDOWREF> alwaysOnTopWindows;
+    std::vector<PHLWINDOWREF> windows;
+    windows.reserve(g_pCompositor->m_vWindows.size());
+
+    for (auto const& w : m_vWindows) {
+        if (w->m_bAlwaysOnTop)
+            alwaysOnTopWindows.emplace_back(w);
+        else
+            windows.emplace_back(w);
+    }
+
+    windows.insert(windows.end(), alwaysOnTopWindows.begin(), alwaysOnTopWindows.end());
+
     auto windowForWorkspace = [&](bool special) -> PHLWINDOW {
         auto floating = [&](bool aboveFullscreen) -> PHLWINDOW {
-            for (auto const& w : m_vWindows | std::views::reverse) {
+            for (auto const& w : windows | std::views::reverse) {
 
                 if (special && !w->onSpecialWorkspace()) // because special floating may creep up into regular
                     continue;
@@ -910,12 +924,12 @@ PHLWINDOW CCompositor::vectorToWindowUnified(const Vector2D& pos, uint8_t proper
                                 // TODO: this is wrong, we should focus the parent, but idk how to get it considering it's nullptr in most cases.
                         }
 
-                        return w;
+                        return w.lock();
                     }
 
                     if (!w->m_bIsX11) {
                         if (w->hasPopupAt(pos))
-                            return w;
+                            return w.lock();
                     }
                 }
             }
@@ -2317,7 +2331,7 @@ void CCompositor::setWindowFullscreenState(const PHLWINDOW PWINDOW, SFullscreenS
 
     // make all windows on the same workspace under the fullscreen window
     for (auto const& w : m_vWindows) {
-        if (w->m_pWorkspace == PWORKSPACE && !w->isFullscreen() && !w->m_bFadingOut && !w->m_bPinned)
+        if (w->m_pWorkspace == PWORKSPACE && !w->isFullscreen() && !w->m_bFadingOut && !w->m_bPinned && !w->m_bAlwaysOnTop)
             w->m_bCreatedOverFullscreen = false;
     }
 
diff --git a/src/debug/HyprCtl.cpp b/src/debug/HyprCtl.cpp
index cd6451e2..e5713c0f 100644
--- a/src/debug/HyprCtl.cpp
+++ b/src/debug/HyprCtl.cpp
@@ -251,6 +251,7 @@ std::string CHyprCtl::getWindowData(PHLWINDOW w, eHyprCtlOutputFormat format) {
     "pid": {},
     "xwayland": {},
     "pinned": {},
+    "alwaysOnTop": {},
     "fullscreen": {},
     "fullscreenClient": {},
     "grouped": [{}],
@@ -264,18 +265,18 @@ std::string CHyprCtl::getWindowData(PHLWINDOW w, eHyprCtlOutputFormat format) {
             escapeJSONStrings(!w->m_pWorkspace ? "" : w->m_pWorkspace->m_szName), ((int)w->m_bIsFloating == 1 ? "true" : "false"), (w->m_bIsPseudotiled ? "true" : "false"),
             (int64_t)w->monitorID(), escapeJSONStrings(w->m_szClass), escapeJSONStrings(w->m_szTitle), escapeJSONStrings(w->m_szInitialClass),
             escapeJSONStrings(w->m_szInitialTitle), w->getPID(), ((int)w->m_bIsX11 == 1 ? "true" : "false"), (w->m_bPinned ? "true" : "false"),
-            (uint8_t)w->m_sFullscreenState.internal, (uint8_t)w->m_sFullscreenState.client, getGroupedData(w, format), getTagsData(w, format),
-            (uintptr_t)w->m_pSwallowed.lock().get(), getFocusHistoryID(w), (g_pInputManager->isWindowInhibiting(w, false) ? "true" : "false"));
+            (w->m_bAlwaysOnTop ? "true" : "false"), (uint8_t)w->m_sFullscreenState.internal, (uint8_t)w->m_sFullscreenState.client, getGroupedData(w, format),
+            getTagsData(w, format), (uintptr_t)w->m_pSwallowed.lock().get(), getFocusHistoryID(w), (g_pInputManager->isWindowInhibiting(w, false) ? "true" : "false"));
     } else {
         return std::format(
             "Window {:x} -> {}:\n\tmapped: {}\n\thidden: {}\n\tat: {},{}\n\tsize: {},{}\n\tworkspace: {} ({})\n\tfloating: {}\n\tpseudo: {}\n\tmonitor: {}\n\tclass: {}\n\ttitle: "
             "{}\n\tinitialClass: {}\n\tinitialTitle: {}\n\tpid: "
-            "{}\n\txwayland: {}\n\tpinned: "
+            "{}\n\txwayland: {}\n\tpinned: {}\n\talwaysOnTop: "
             "{}\n\tfullscreen: {}\n\tfullscreenClient: {}\n\tgrouped: {}\n\ttags: {}\n\tswallowing: {:x}\n\tfocusHistoryID: {}\n\tinhibitingIdle: {}\n\n",
             (uintptr_t)w.get(), w->m_szTitle, (int)w->m_bIsMapped, (int)w->isHidden(), (int)w->m_vRealPosition->goal().x, (int)w->m_vRealPosition->goal().y,
             (int)w->m_vRealSize->goal().x, (int)w->m_vRealSize->goal().y, w->m_pWorkspace ? w->workspaceID() : WORKSPACE_INVALID,
             (!w->m_pWorkspace ? "" : w->m_pWorkspace->m_szName), (int)w->m_bIsFloating, (int)w->m_bIsPseudotiled, (int64_t)w->monitorID(), w->m_szClass, w->m_szTitle,
-            w->m_szInitialClass, w->m_szInitialTitle, w->getPID(), (int)w->m_bIsX11, (int)w->m_bPinned, (uint8_t)w->m_sFullscreenState.internal,
+            w->m_szInitialClass, w->m_szInitialTitle, w->getPID(), (int)w->m_bIsX11, (int)w->m_bPinned, (int)w->m_bAlwaysOnTop, (uint8_t)w->m_sFullscreenState.internal,
             (uint8_t)w->m_sFullscreenState.client, getGroupedData(w, format), getTagsData(w, format), (uintptr_t)w->m_pSwallowed.lock().get(), getFocusHistoryID(w),
             (int)g_pInputManager->isWindowInhibiting(w, false));
     }
diff --git a/src/desktop/Window.hpp b/src/desktop/Window.hpp
index ce2f8eb2..7e0237eb 100644
--- a/src/desktop/Window.hpp
+++ b/src/desktop/Window.hpp
@@ -318,6 +318,9 @@ class CWindow {
     // For pinned (sticky) windows
     bool m_bPinned = false;
 
+    // For placing above other windows in the workspace
+    bool m_bAlwaysOnTop = false;
+
     // For preserving pinned state when fullscreening a pinned window
     bool m_bPinFullscreened = false;
 
diff --git a/src/desktop/WindowRule.cpp b/src/desktop/WindowRule.cpp
index 306a25ce..338a5da8 100644
--- a/src/desktop/WindowRule.cpp
+++ b/src/desktop/WindowRule.cpp
@@ -5,7 +5,7 @@
 #include "../config/ConfigManager.hpp"
 
 static const auto RULES = std::unordered_set<std::string>{
-    "float", "fullscreen", "maximize", "noinitialfocus", "pin", "stayfocused", "tile", "renderunfocused",
+    "float", "fullscreen", "maximize", "noinitialfocus", "pin", "alwaysontop", "stayfocused", "tile", "renderunfocused",
 };
 static const auto RULES_PREFIX = std::unordered_set<std::string>{
     "animation", "bordercolor", "bordersize", "center",   "fullscreenstate", "group",       "idleinhibit",    "maxsize", "minsize",       "monitor", "move",      "opacity",
@@ -32,6 +32,8 @@ CWindowRule::CWindowRule(const std::string& rule, const std::string& value, bool
         ruleType = RULE_NOINITIALFOCUS;
     else if (rule == "pin")
         ruleType = RULE_PIN;
+    else if (rule == "alwaysontop")
+        ruleType = RULE_ALWAYSONTOP;
     else if (rule == "stayfocused")
         ruleType = RULE_STAYFOCUSED;
     else if (rule == "tile")
diff --git a/src/desktop/WindowRule.hpp b/src/desktop/WindowRule.hpp
index be9c2d9c..ed2b1c5e 100644
--- a/src/desktop/WindowRule.hpp
+++ b/src/desktop/WindowRule.hpp
@@ -15,6 +15,7 @@ class CWindowRule {
         RULE_MAXIMIZE,
         RULE_NOINITIALFOCUS,
         RULE_PIN,
+        RULE_ALWAYSONTOP,
         RULE_STAYFOCUSED,
         RULE_TILE,
         RULE_RENDERUNFOCUSED,
diff --git a/src/events/Windows.cpp b/src/events/Windows.cpp
index f6ef9ced..863bf021 100644
--- a/src/events/Windows.cpp
+++ b/src/events/Windows.cpp
@@ -246,6 +246,10 @@ void Events::listener_mapWindow(void* owner, void* data) {
                 PWINDOW->m_bPinned = true;
                 break;
             }
+            case CWindowRule::RULE_ALWAYSONTOP: {
+                PWINDOW->m_bAlwaysOnTop = true;
+                break;
+            }
             case CWindowRule::RULE_FULLSCREEN: {
                 requestedInternalFSMode = FSMODE_FULLSCREEN;
                 break;
@@ -317,6 +321,9 @@ void Events::listener_mapWindow(void* owner, void* data) {
     if (PWINDOW->m_bPinned && !PWINDOW->m_bIsFloating)
         PWINDOW->m_bPinned = false;
 
+    if (PWINDOW->m_bAlwaysOnTop && !PWINDOW->m_bIsFloating)
+        PWINDOW->m_bAlwaysOnTop = false;
+
     const CVarList WORKSPACEARGS = CVarList(requestedWorkspace, 0, ' ');
 
     if (!WORKSPACEARGS[0].empty()) {
diff --git a/src/layout/IHyprLayout.cpp b/src/layout/IHyprLayout.cpp
index 9a0ba6ee..9031e12f 100644
--- a/src/layout/IHyprLayout.cpp
+++ b/src/layout/IHyprLayout.cpp
@@ -713,7 +713,8 @@ void IHyprLayout::changeWindowFloatingMode(PHLWINDOW pWindow) {
         g_pCompositor->setWindowFullscreenInternal(pWindow, FSMODE_NONE);
     }
 
-    pWindow->m_bPinned = false;
+    pWindow->m_bPinned      = false;
+    pWindow->m_bAlwaysOnTop = false;
 
     g_pHyprRenderer->damageWindow(pWindow, true);
 
diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index 247550ae..a7c7bf2f 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -125,6 +125,7 @@ CKeybindManager::CKeybindManager() {
     m_mDispatchers["swapnext"]                       = swapnext;
     m_mDispatchers["swapactiveworkspaces"]           = swapActiveWorkspaces;
     m_mDispatchers["pin"]                            = pinActive;
+    m_mDispatchers["togglealwaysontop"]              = toggleActiveAlwaysOnTop;
     m_mDispatchers["mouse"]                          = mouse;
     m_mDispatchers["bringactivetotop"]               = bringActiveToTop;
     m_mDispatchers["alterzorder"]                    = alterZOrder;
@@ -2700,6 +2701,31 @@ SDispatchResult CKeybindManager::pinActive(std::string args) {
     return {};
 }
 
+SDispatchResult CKeybindManager::toggleActiveAlwaysOnTop(std::string args) {
+
+    PHLWINDOW PWINDOW = nullptr;
+
+    if (args != "active" && args.length() > 1)
+        PWINDOW = g_pCompositor->getWindowByRegex(args);
+    else
+        PWINDOW = g_pCompositor->m_pLastWindow.lock();
+
+    if (!PWINDOW) {
+        Debug::log(ERR, "togglealwaysontop: window not found");
+        return {.success = false, .error = "togglealwaysontop: window not found"};
+    }
+
+    if (!PWINDOW->m_bIsFloating || PWINDOW->isFullscreen())
+        return {};
+
+    PWINDOW->m_bAlwaysOnTop = !PWINDOW->m_bAlwaysOnTop;
+
+    PWINDOW->updateDynamicRules();
+    g_pCompositor->updateWindowAnimatedDecorationValues(PWINDOW);
+
+    return {};
+}
+
 SDispatchResult CKeybindManager::mouse(std::string args) {
     const auto ARGS    = CVarList(args.substr(1), 2, ' ');
     const auto PRESSED = args[0] == '1';
diff --git a/src/managers/KeybindManager.hpp b/src/managers/KeybindManager.hpp
index 1848ca78..07e44be2 100644
--- a/src/managers/KeybindManager.hpp
+++ b/src/managers/KeybindManager.hpp
@@ -207,6 +207,7 @@ class CKeybindManager {
     static SDispatchResult swapnext(std::string);
     static SDispatchResult swapActiveWorkspaces(std::string);
     static SDispatchResult pinActive(std::string);
+    static SDispatchResult toggleActiveAlwaysOnTop(std::string);
     static SDispatchResult mouse(std::string);
     static SDispatchResult bringActiveToTop(std::string);
     static SDispatchResult alterZOrder(std::string);
diff --git a/src/render/Renderer.cpp b/src/render/Renderer.cpp
index 42445728..7b9f3ed3 100644
--- a/src/render/Renderer.cpp
+++ b/src/render/Renderer.cpp
@@ -338,8 +338,8 @@ void CHyprRenderer::renderWorkspaceWindowsFullscreen(PHLMONITOR pMonitor, PHLWOR
 
     // then render windows over fullscreen.
     for (auto const& w : g_pCompositor->m_vWindows) {
-        if (w->m_pWorkspace != pWorkspaceWindow->m_pWorkspace || !w->m_bIsFloating || (!w->m_bCreatedOverFullscreen && !w->m_bPinned) || (!w->m_bIsMapped && !w->m_bFadingOut) ||
-            w->isFullscreen())
+        if (w->m_pWorkspace != pWorkspaceWindow->m_pWorkspace || !w->m_bIsFloating || (!w->m_bCreatedOverFullscreen && !w->m_bPinned && !w->m_bAlwaysOnTop) ||
+            (!w->m_bIsMapped && !w->m_bFadingOut) || w->isFullscreen())
             continue;
 
         if (w->m_pMonitor == pWorkspace->m_pMonitor && pWorkspace->m_bIsSpecialWorkspace != w->onSpecialWorkspace())
@@ -412,6 +412,30 @@ void CHyprRenderer::renderWorkspaceWindows(PHLMONITOR pMonitor, PHLWORKSPACE pWo
         if (!w)
             continue;
 
+        if (w->m_bAlwaysOnTop)
+            continue;
+
+        if (!w->m_bIsFloating || w->m_bPinned)
+            continue;
+
+        if (pWorkspace->m_bIsSpecialWorkspace != w->onSpecialWorkspace())
+            continue;
+
+        if (pWorkspace->m_bIsSpecialWorkspace && w->m_pMonitor != pWorkspace->m_pMonitor)
+            continue; // special on another are rendered as a part of the base pass
+
+        // render the bad boy
+        renderWindow(w.lock(), pMonitor, time, true, RENDER_PASS_ALL);
+    }
+
+    // always on top floating windows
+    for (auto& w : windows) {
+        if (!w)
+            continue;
+
+        if (!w->m_bAlwaysOnTop)
+            continue;
+
         if (!w->m_bIsFloating || w->m_bPinned)
             continue;
 
