diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index 1bd7e1a2..e9ad6682 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -2117,13 +2117,13 @@ SDispatchResult CKeybindManager::toggleSpecialWorkspace(std::string args) {
         return {.success = false, .error = "Invalid workspace passed to special"};
     }
 
-    bool       requestedWorkspaceIsAlreadyOpen = false;
-    const auto PMONITOR                        = g_pCompositor->m_lastMonitor;
-    auto       specialOpenOnMonitor            = PMONITOR->activeSpecialWorkspaceID();
+    bool requestedWorkspaceIsAlreadyOpen = false;
+    auto PMONITOR                        = g_pCompositor->m_lastMonitor.get();
 
     for (auto const& m : g_pCompositor->m_monitors) {
         if (m->activeSpecialWorkspaceID() == workspaceID) {
             requestedWorkspaceIsAlreadyOpen = true;
+            PMONITOR                        = m.get();
             break;
         }
     }
@@ -2132,8 +2132,7 @@ SDispatchResult CKeybindManager::toggleSpecialWorkspace(std::string args) {
 
     PHLWORKSPACEREF focusedWorkspace;
 
-    if (requestedWorkspaceIsAlreadyOpen && specialOpenOnMonitor == workspaceID) {
-        // already open on this monitor
+    if (requestedWorkspaceIsAlreadyOpen) {
         Debug::log(LOG, "Toggling special workspace {} to closed", workspaceID);
         PMONITOR->setSpecialWorkspace(nullptr);
 
