diff --git a/src/Compositor.cpp b/src/Compositor.cpp
index 844a19f9..f344e15e 100644
--- a/src/Compositor.cpp
+++ b/src/Compositor.cpp
@@ -2459,15 +2459,15 @@ PHLLS CCompositor::getLayerSurfaceFromSurface(SP<CWLSurfaceResource> pSurface) {
 }
 
 // returns a delta
-Vector2D CCompositor::parseWindowVectorArgsRelative(const std::string& args, const Vector2D& relativeTo) {
+Vector2D CCompositor::parseWindowVectorArgsRelative(const std::string& args, const PHLWINDOW pWindow, const bool isPosition) {
+    const auto& relativeTo = isPosition ? pWindow->m_realPosition->goal() : pWindow->m_realSize->goal();
+
     if (!args.contains(' ') && !args.contains('\t'))
         return relativeTo;
 
-    const auto PMONITOR = Desktop::focusState()->monitor();
-
-    bool       xIsPercent = false;
-    bool       yIsPercent = false;
-    bool       isExact    = false;
+    bool        xIsPercent = false;
+    bool        yIsPercent = false;
+    bool        isExact    = false;
 
     CVarList2  varList(std::string{args}, 0, 's', true);
     auto       x = varList[0];
@@ -2498,8 +2498,17 @@ Vector2D CCompositor::parseWindowVectorArgsRelative(const std::string& args, con
     int Y = 0;
 
     if (isExact) {
+        const auto PMONITOR = pWindow->m_monitor.lock();
+        if (!PMONITOR)
+            return relativeTo;
+
         X = xIsPercent ? *strToNumber<float>(x) * 0.01 * PMONITOR->m_size.x : *strToNumber<int64_t>(x);
         Y = yIsPercent ? *strToNumber<float>(y) * 0.01 * PMONITOR->m_size.y : *strToNumber<int64_t>(y);
+
+        if (isPosition) {
+            X += xIsPercent ? PMONITOR->m_position.x : 0;
+            Y += xIsPercent ? PMONITOR->m_position.y : 0;
+        }
     } else {
         X = xIsPercent ? (*strToNumber<float>(x) * 0.01 * relativeTo.x) + relativeTo.x : *strToNumber<int64_t>(x) + relativeTo.x;
         Y = yIsPercent ? (*strToNumber<float>(y) * 0.01 * relativeTo.y) + relativeTo.y : *strToNumber<int64_t>(y) + relativeTo.y;
diff --git a/src/Compositor.hpp b/src/Compositor.hpp
index 96850407..124ae43f 100644
--- a/src/Compositor.hpp
+++ b/src/Compositor.hpp
@@ -144,7 +144,7 @@ class CCompositor {
     void                   warpCursorTo(const Vector2D&, bool force = false);
     PHLLS                  getLayerSurfaceFromSurface(SP<CWLSurfaceResource>);
     void                   closeWindow(PHLWINDOW);
-    Vector2D               parseWindowVectorArgsRelative(const std::string&, const Vector2D&);
+    Vector2D               parseWindowVectorArgsRelative(const std::string&, const PHLWINDOW pWindow, const bool isPosition);
     [[nodiscard]] PHLWORKSPACE          createNewWorkspace(const WORKSPACEID&, const MONITORID&, const std::string& name = "",
                                                            bool isEmpty = true); // will be deleted next frame if left empty and unfocused!
     bool                                isWorkspaceSpecial(const WORKSPACEID&);
diff --git a/src/managers/KeybindManager.cpp b/src/managers/KeybindManager.cpp
index 4a698c16..c7658a6e 100644
--- a/src/managers/KeybindManager.cpp
+++ b/src/managers/KeybindManager.cpp
@@ -1915,7 +1915,7 @@ SDispatchResult CKeybindManager::resizeActive(std::string args) {
     if (PLASTWINDOW->isFullscreen())
         return {.success = false, .error = "Window is fullscreen"};
 
-    const auto SIZ = g_pCompositor->parseWindowVectorArgsRelative(args, PLASTWINDOW->m_realSize->goal());
+    const auto SIZ = g_pCompositor->parseWindowVectorArgsRelative(args, PLASTWINDOW, false);
 
     if (SIZ.x < 1 || SIZ.y < 1)
         return {.success = false, .error = "Invalid size provided"};
@@ -1937,7 +1937,7 @@ SDispatchResult CKeybindManager::moveActive(std::string args) {
     if (PLASTWINDOW->isFullscreen())
         return {.success = false, .error = "Window is fullscreen"};
 
-    const auto POS = g_pCompositor->parseWindowVectorArgsRelative(args, PLASTWINDOW->m_realPosition->goal());
+    const auto POS = g_pCompositor->parseWindowVectorArgsRelative(args, PLASTWINDOW, true);
 
     g_layoutManager->moveTarget(POS - PLASTWINDOW->m_realPosition->goal(), PLASTWINDOW->layoutTarget());
 
@@ -1959,7 +1959,7 @@ SDispatchResult CKeybindManager::moveWindow(std::string args) {
     if (PWINDOW->isFullscreen())
         return {.success = false, .error = "Window is fullscreen"};
 
-    const auto POS = g_pCompositor->parseWindowVectorArgsRelative(MOVECMD, PWINDOW->m_realPosition->goal());
+    const auto POS = g_pCompositor->parseWindowVectorArgsRelative(MOVECMD, PWINDOW, true);
 
     g_layoutManager->moveTarget(POS - PWINDOW->m_realPosition->goal(), PWINDOW->layoutTarget());
 
@@ -1981,7 +1981,7 @@ SDispatchResult CKeybindManager::resizeWindow(std::string args) {
     if (PWINDOW->isFullscreen())
         return {.success = false, .error = "Window is fullscreen"};
 
-    const auto SIZ = g_pCompositor->parseWindowVectorArgsRelative(MOVECMD, PWINDOW->m_realSize->goal());
+    const auto SIZ = g_pCompositor->parseWindowVectorArgsRelative(MOVECMD, PWINDOW, false);
 
     if (SIZ.x < 1 || SIZ.y < 1)
         return {.success = false, .error = "Invalid size provided"};
