diff --git a/src/config/ConfigManager.cpp b/src/config/ConfigManager.cpp
index ac5c101e..d7f94b1b 100644
--- a/src/config/ConfigManager.cpp
+++ b/src/config/ConfigManager.cpp
@@ -1977,7 +1977,9 @@ std::optional<std::string> CConfigManager::handleBind(const std::string& command
     const auto DISPATCHER = g_pKeybindManager->m_mDispatchers.find(HANDLER);
 
     if (DISPATCHER == g_pKeybindManager->m_mDispatchers.end()) {
-        Debug::log(ERR, "Invalid dispatcher!");
+        // Do not complain about missing plugin dispatchers
+        if (!HANDLER.contains(":"))
+            Debug::log(ERR, "Invalid dispatcher!");
         return "Invalid dispatcher, requested \"" + HANDLER + "\" does not exist";
     }
 
