diff --git a/src/desktop/view/Window.cpp b/src/desktop/view/Window.cpp
index 32a0086e..3fa78cb8 100644
--- a/src/desktop/view/Window.cpp
+++ b/src/desktop/view/Window.cpp
@@ -2173,6 +2173,7 @@ void CWindow::mapWindow() {
 
             m_workspace = pWorkspace;
             m_monitor   = pWorkspace->m_monitor;
+            PMONITOR    = m_monitor.lock();
 
             if (m_monitor.lock()->m_activeSpecialWorkspace && !pWorkspace->m_isSpecialWorkspace)
                 workspaceSilent = true;
@@ -2182,8 +2183,6 @@ void CWindow::mapWindow() {
                     pWorkspace->m_monitor->setSpecialWorkspace(pWorkspace);
                 else if (PMONITOR->activeWorkspaceID() != requestedWorkspaceID && !m_noInitialFocus)
                     g_pKeybindManager->m_dispatchers["workspace"](requestedWorkspaceName);
-
-                PMONITOR = Desktop::focusState()->monitor();
             }
 
             requestedFSMonitor = MONITOR_INVALID;
