diff --git a/src/managers/input/InputManager.cpp b/src/managers/input/InputManager.cpp
index 1d16f915..a6e75ca4 100644
--- a/src/managers/input/InputManager.cpp
+++ b/src/managers/input/InputManager.cpp
@@ -170,6 +170,7 @@ void CInputManager::mouseMoveUnified(uint32_t time, bool refocus, bool mouse) {
     static auto PRESIZEONBORDER       = CConfigValue<Hyprlang::INT>("general:resize_on_border");
     static auto PRESIZECURSORICON     = CConfigValue<Hyprlang::INT>("general:hover_icon_on_border");
     static auto PZOOMFACTOR           = CConfigValue<Hyprlang::FLOAT>("cursor:zoom_factor");
+    static auto PSPECIALFALLTHROUGH   = CConfigValue<Hyprlang::INT>("input:special_fallthrough");
 
     const auto  FOLLOWMOUSE = *PFOLLOWONDND && PROTO::data->dndActive() ? 1 : *PFOLLOWMOUSE;
 
@@ -312,7 +313,8 @@ void CInputManager::mouseMoveUnified(uint32_t time, bool refocus, bool mouse) {
     // then, we check if the workspace doesnt have a fullscreen window
     const auto PWORKSPACE   = PMONITOR->activeWorkspace;
     const auto PWINDOWIDEAL = g_pCompositor->vectorToWindowUnified(mouseCoords, RESERVED_EXTENTS | INPUT_EXTENTS | ALLOW_FLOATING);
-    if (PWORKSPACE->m_bHasFullscreenWindow && !foundSurface && PWORKSPACE->m_efFullscreenMode == FSMODE_FULLSCREEN) {
+    if (PWORKSPACE->m_bHasFullscreenWindow && !foundSurface && PWORKSPACE->m_efFullscreenMode == FSMODE_FULLSCREEN &&
+        (!*PSPECIALFALLTHROUGH && !PMONITOR->activeSpecialWorkspace)) {
         pFoundWindow = PWORKSPACE->getFullscreenWindow();
 
         if (!pFoundWindow) {
@@ -321,9 +323,7 @@ void CInputManager::mouseMoveUnified(uint32_t time, bool refocus, bool mouse) {
             return;
         }
 
-        if (PWINDOWIDEAL &&
-            ((PWINDOWIDEAL->m_bIsFloating && PWINDOWIDEAL->m_bCreatedOverFullscreen) /* floating over fullscreen */
-             || (PMONITOR->activeSpecialWorkspace == PWINDOWIDEAL->m_pWorkspace) /* on an open special workspace */))
+        if (PWINDOWIDEAL && PWINDOWIDEAL->m_bIsFloating && PWINDOWIDEAL->m_bCreatedOverFullscreen)
             pFoundWindow = PWINDOWIDEAL;
 
         if (!pFoundWindow->m_bIsX11) {
