From 3c040f8a5069e0a6aa55be04670027f484f17ddb Mon Sep 17 00:00:00 2001
From: gnusenpai <pspslimboy98@gmail.com>
Date: Thu, 3 Mar 2022 17:43:40 -0500
Subject: [PATCH] fix read_cfg with legacy_layout disabled

---
 src/overlay_params.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/overlay_params.cpp b/src/overlay_params.cpp
index 5818c45..2ebf9d7 100644
--- a/src/overlay_params.cpp
+++ b/src/overlay_params.cpp
@@ -697,7 +697,7 @@ parse_overlay_config(struct overlay_params *params,
 
    // TODO decide what to do for legacy_layout=0
    // second pass, override config file settings with MANGOHUD_CONFIG
-   if (params->enabled[OVERLAY_PARAM_ENABLED_legacy_layout] && env && read_cfg) {
+   if (env && read_cfg) {
       // If passing legacy_layout=0 to MANGOHUD_CONFIG anyway then clear first pass' results
       HUDElements.ordered_functions.clear();
       parse_overlay_env(params, env);
-- 
2.34.1
