diff --git a/core/agents/lastfm/agent.go b/core/agents/lastfm/agent.go
index ec732f17..62b741c8 100644
--- a/core/agents/lastfm/agent.go
+++ b/core/agents/lastfm/agent.go
@@ -294,8 +294,8 @@ func (l *lastfmAgent) NowPlaying(ctx context.Context, userId string, track *mode
 
 	err = l.client.updateNowPlaying(ctx, sk, ScrobbleInfo{
 		artist:      l.getArtistForScrobble(track),
-		track:       track.Title,
-		album:       track.Album,
+		track:       strings.ReplaceAll(track.Title, "’" , "'"),
+		album:       strings.ReplaceAll(track.Album, "’" , "'"),
 		trackNumber: track.TrackNumber,
 		mbid:        track.MbzRecordingID,
 		duration:    int(track.Duration),
@@ -320,8 +320,8 @@ func (l *lastfmAgent) Scrobble(ctx context.Context, userId string, s scrobbler.S
 	}
 	err = l.client.scrobble(ctx, sk, ScrobbleInfo{
 		artist:      l.getArtistForScrobble(&s.MediaFile),
-		track:       s.Title,
-		album:       s.Album,
+		track:       strings.ReplaceAll(s.Title, "’" , "'"),
+		album:       strings.ReplaceAll(s.Album, "’" , "'"),
 		trackNumber: s.TrackNumber,
 		mbid:        s.MbzRecordingID,
 		duration:    int(s.Duration),
