diff --git a/networkd-dispatcher b/networkd-dispatcher
index 8ed2269..7518390 100755
--- a/networkd-dispatcher
+++ b/networkd-dispatcher
@@ -20,6 +20,7 @@ import re
 import socket
 import subprocess
 import sys
+import time
 
 # Try to import the dynamic glib, or try to fall back to static
 try:
@@ -270,6 +271,7 @@ class Dispatcher():
         return f"<Dispatcher({self.__dict__})>"
 
     def _interface_scan(self):
+        time.sleep(2)
         iface_list = get_networkctl_list()
         # Append new interfaces, keeping old ones around to avoid hotplug race
         # condition (issue #20)
