diff --git a/src/nixos-anywhere.sh b/src/nixos-anywhere.sh
index a72c5bf..3fb48fe 100755
--- a/src/nixos-anywhere.sh
+++ b/src/nixos-anywhere.sh
@@ -317,6 +317,7 @@ nixBuild() {
   NIX_SSHOPTS="-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i $sshKeyDir/nixos-anywhere ${sshArgs[*]}" nix build \
     --print-out-paths \
     --no-link \
+    --override-input firstBoot "github:JManch/true" \
     "${nixOptions[@]}" \
     "$@"
 }
@@ -551,7 +552,7 @@ nixosInstall() {
 
   if [[ -n ${extraFiles} ]]; then
     step Copying extra files
-    tar -C "$extraFiles" -cpf- . | runSsh "${maybeSudo} tar -C /mnt -xf- --no-same-owner"
+    sudo tar -C "$extraFiles" -cpf- . | runSsh "${maybeSudo} tar -C /mnt -xf- --same-owner"
     runSsh "chmod 755 /mnt" # tar also changes permissions of /mnt
   fi
 
