diff --git a/src/nixos-anywhere.sh b/src/nixos-anywhere.sh
index 9010ae6..82daf69 100755
--- a/src/nixos-anywhere.sh
+++ b/src/nixos-anywhere.sh
@@ -218,6 +218,7 @@ nix_build() {
   NIX_SSHOPTS="-o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i $ssh_key_dir/nixos-anywhere ${ssh_args[*]}" nix build \
     --print-out-paths \
     --no-link \
+    --override-input firstBoot "github:JManch/true" \
     "${nix_options[@]}" \
     "$@"
 }
@@ -484,7 +485,7 @@ fi
 
 if [[ -n ${extra_files-} ]]; then
   step Copying extra files
-  tar -C "$extra_files" -cpf- . | ssh_ "${maybe_sudo} tar -C /mnt -xf- --no-same-owner"
+  tar -C "$extra_files" -cpf- . | ssh_ "${maybe_sudo} tar -C /mnt -xf- --same-owner"
   ssh_ "chmod 755 /mnt" # tar also changes permissions of /mnt
 fi
 
