diff --git a/spotify_player/src/client/mod.rs b/spotify_player/src/client/mod.rs
index d911bd8..87f4362 100644
--- a/spotify_player/src/client/mod.rs
+++ b/spotify_player/src/client/mod.rs
@@ -1530,7 +1530,9 @@ impl Client {
             ))
             .body(&get_text_from_format_str(
                 &configs.app_config.notify_format.body,
-            ));
+            ))
+            .urgency(notify_rust::Urgency::Low)
+            .hint(notify_rust::Hint::Custom("x-canonical-private-synchronous".to_string(),"anything".to_string()));
         if cover_img_path.exists() {
             n.icon(cover_img_path.to_str().context("valid cover_img_path")?);
         }
