diff --git a/src/modules/cpu_usage/common.cpp b/src/modules/cpu_usage/common.cpp
index 13e9cc4e..a1b65407 100644
--- a/src/modules/cpu_usage/common.cpp
+++ b/src/modules/cpu_usage/common.cpp
@@ -77,6 +77,7 @@ std::tuple<std::vector<uint16_t>, std::string> waybar::modules::CpuUsage::getCpu
       const float delta_total = curr_total - prev_total;
       uint16_t tmp =
           (delta_total > 0) ? static_cast<uint16_t>(100 * (1 - delta_idle / delta_total)) : 0;
+      if (tmp <= 3) tmp = 0;
       tooltip = fmt::format("Total: {}%\nCores: (pending)", tmp);
       usage.push_back(tmp);
     } else {
@@ -100,6 +101,7 @@ std::tuple<std::vector<uint16_t>, std::string> waybar::modules::CpuUsage::getCpu
     const float delta_total = curr_total - prev_total;
     uint16_t tmp =
         (delta_total > 0) ? static_cast<uint16_t>(100 * (1 - delta_idle / delta_total)) : 0;
+    if (tmp <= 3) tmp = 0;
     if (i == 0) {
       tooltip = fmt::format("Total: {}%", tmp);
     } else {
