diff --git a/src/modules/wireplumber.cpp b/src/modules/wireplumber.cpp
index a43ad29b..f24a02da 100644
--- a/src/modules/wireplumber.cpp
+++ b/src/modules/wireplumber.cpp
@@ -79,7 +79,8 @@ void waybar::modules::Wireplumber::updateNodeName(waybar::modules::Wireplumber*
   if (proxy == nullptr) {
     auto err = fmt::format("Object '{}' not found\n", id);
     spdlog::error("[{}]: {}", self->name_, err);
-    throw std::runtime_error(err);
+    self->node_name_ = "";
+    return;
   }
 
   g_autoptr(WpProperties) properties =
@@ -90,12 +91,9 @@ void waybar::modules::Wireplumber::updateNodeName(waybar::modules::Wireplumber*
   properties = wp_properties_ensure_unique_owner(properties);
   wp_properties_add(properties, globalP);
   wp_properties_set(properties, "object.id", nullptr);
-  const auto* nick = wp_properties_get(properties, "node.nick");
   const auto* description = wp_properties_get(properties, "node.description");
 
-  self->node_name_ = nick != nullptr          ? nick
-                     : description != nullptr ? description
-                                              : "Unknown node name";
+  self->node_name_ = description != nullptr ? description : "Unknown node name";
   spdlog::debug("[{}]: Updating '{}' node name to: {}", self->name_, self->type_, self->node_name_);
 }
 
@@ -192,18 +190,6 @@ void waybar::modules::Wireplumber::onMixerChanged(waybar::modules::Wireplumber*
       self->om_, WP_TYPE_NODE, WP_CONSTRAINT_TYPE_G_PROPERTY, "bound-id", "=u", id, nullptr));
 
   if (node == nullptr) {
-    // log a warning only if no other widget is targeting the id.
-    // this reduces log spam when multiple instances of the module are used on different node types.
-    if (id != self->node_id_ && id != self->source_node_id_) {
-      for (auto const& module : waybar::modules::Wireplumber::modules) {
-        if (module->node_id_ == id || module->source_node_id_ == id) {
-          return;
-        }
-      }
-    }
-
-    spdlog::warn("[{}]: (onMixerChanged: {}) - Object with id {} not found", self->name_,
-                 self->type_, id);
     return;
   }
 
@@ -436,6 +422,12 @@ auto waybar::modules::Wireplumber::update() -> void {
     label_.get_style_context()->remove_class("source-muted");
   }
 
+  if (format.empty() || node_name_.empty()) {
+    event_box_.hide();
+    return;
+  }
+  event_box_.show();
+
   int vol = round(volume_ * 100.0);
   int source_vol = round(source_volume_ * 100.0);
 
