diff --git a/src/modules/wireplumber.cpp b/src/modules/wireplumber.cpp
index 106ca403..9f2bd2b7 100644
--- a/src/modules/wireplumber.cpp
+++ b/src/modules/wireplumber.cpp
@@ -74,7 +74,8 @@ void waybar::modules::Wireplumber::updateNodeName(waybar::modules::Wireplumber*
   if (proxy == nullptr) {
     auto err = fmt::format("Object '{}' not found\n", id);
     spdlog::error("[{}]: {}", self->name_, err);
-    throw std::runtime_error(err);
+    self->node_name_ = "";
+    return;
   }
 
   g_autoptr(WpProperties) properties =
@@ -85,12 +86,9 @@ void waybar::modules::Wireplumber::updateNodeName(waybar::modules::Wireplumber*
   properties = wp_properties_ensure_unique_owner(properties);
   wp_properties_add(properties, globalP);
   wp_properties_set(properties, "object.id", nullptr);
-  const auto* nick = wp_properties_get(properties, "node.nick");
   const auto* description = wp_properties_get(properties, "node.description");
 
-  self->node_name_ = nick != nullptr          ? nick
-                     : description != nullptr ? description
-                                              : "Unknown node name";
+  self->node_name_ = description != nullptr ? description : "Unknown node name";
   spdlog::debug("[{}]: Updating '{}' node name to: {}", self->name_, self->type_, self->node_name_);
 }
 
@@ -339,6 +337,12 @@ auto waybar::modules::Wireplumber::update() -> void {
     label_.get_style_context()->remove_class("muted");
   }
 
+  if (format.empty() || node_name_.empty()) {
+    event_box_.hide();
+    return;
+  }
+  event_box_.show();
+
   int vol = round(volume_ * 100.0);
   std::string markup = fmt::format(fmt::runtime(format), fmt::arg("node_name", node_name_),
                                    fmt::arg("volume", vol), fmt::arg("icon", getIcon(vol)));
