diff --git a/src/main.cpp b/src/main.cpp
index 045b2cd4..2cee774b 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -81,9 +81,6 @@ int main(int argc, char* argv[]) {
     });
 
     std::signal(SIGUSR2, [](int /*signal*/) {
-      spdlog::info("Reloading...");
-      reload = true;
-      waybar::Client::inst()->reset();
     });
 
     std::signal(SIGINT, [](int /*signal*/) {
diff --git a/src/modules/hyprland/workspaces.cpp b/src/modules/hyprland/workspaces.cpp
index ef057d6d..f9a701e5 100644
--- a/src/modules/hyprland/workspaces.cpp
+++ b/src/modules/hyprland/workspaces.cpp
@@ -674,7 +674,6 @@ auto Workspaces::registerIpc() -> void {
   m_ipc.registerForIPC("closewindow", this);
   m_ipc.registerForIPC("movewindow", this);
   m_ipc.registerForIPC("urgent", this);
-  m_ipc.registerForIPC("configreloaded", this);
 
   if (windowRewriteConfigUsesTitle()) {
     spdlog::info(
