diff --git a/src/main.cpp b/src/main.cpp
index ff446ffc..9989f012 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -93,9 +93,10 @@ int main(int argc, char* argv[]) {
 
     for (int sig = SIGRTMIN + 1; sig <= SIGRTMAX; ++sig) {
       std::signal(sig, [](int sig) {
-        for (auto& bar : waybar::Client::inst()->bars) {
-          bar->handleSignal(sig);
-        }
+        int monitorNum = sig - SIGRTMIN - 1;
+        auto& bars = waybar::Client::inst()->bars;
+        if (monitorNum < bars.size())
+          bars.at(monitorNum).get()->toggle();
       });
     }
     startSignalThread();
