diff --git a/src/modules/custom.cpp b/src/modules/custom.cpp
index 836a0d71..28013f25 100644
--- a/src/modules/custom.cpp
+++ b/src/modules/custom.cpp
@@ -123,7 +123,6 @@ void waybar::modules::Custom::handleContinuousProcessExit(int exit_code) {
   if (exit_code != 0) {
     output_ = {.exit_code = exit_code, .out = ""};
     dp.emit();
-    spdlog::error("{} stopped unexpectedly, is it endless?", name_);
   }
 
   scheduleContinuousRestart();
