diff --git a/Makefile.in b/Makefile.in
index 1028f97be..36628a58c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -183,28 +183,6 @@ install: all
 	$(INSTALL) -m 0700 extras/unrealircd-upgrade-script $(DESTDIR)@BINDIR@/unrealircd-upgrade-script
 	$(INSTALL) -m 0700 -d $(DESTDIR)@DOCDIR@
 	$(INSTALL) -m 0600 doc/Authors doc/coding-guidelines doc/tao.of.irc doc/KEYS doc/RELEASE-NOTES.md $(DESTDIR)@DOCDIR@
-	$(INSTALL) -m 0700 -d $(DESTDIR)@CONFDIR@
-	$(INSTALL) -m 0600 doc/conf/*.default.conf $(DESTDIR)@CONFDIR@
-	$(INSTALL) -m 0600 doc/conf/*.optional.conf $(DESTDIR)@CONFDIR@
-	-@if [ ! -f "$(DESTDIR)@CONFDIR@/modules.sources.list" ] ; then \
-		$(INSTALL) -m 0600 doc/conf/modules.sources.list $(DESTDIR)@CONFDIR@ ; \
-	fi
-	-@if [ ! -f "$(DESTDIR)@CONFDIR@/spamfilter.conf" ] ; then \
-		$(INSTALL) -m 0600 doc/conf/spamfilter.conf $(DESTDIR)@CONFDIR@ ; \
-	fi
-	-@extras/patches/patch_spamfilter_conf "$(DESTDIR)@CONFDIR@"
-	-@if [ ! -f "$(DESTDIR)@CONFDIR@/badwords.conf" ] ; then \
-		$(INSTALL) -m 0600 doc/conf/badwords.conf $(DESTDIR)@CONFDIR@ ; \
-	fi
-	-@if [ ! -f "$(DESTDIR)@CONFDIR@/dccallow.conf" ] ; then \
-		$(INSTALL) -m 0600 doc/conf/dccallow.conf $(DESTDIR)@CONFDIR@ ; \
-	fi
-	$(INSTALL) -m 0700 -d $(DESTDIR)@CONFDIR@/aliases
-	$(INSTALL) -m 0600 doc/conf/aliases/*.conf $(DESTDIR)@CONFDIR@/aliases
-	$(INSTALL) -m 0700 -d $(DESTDIR)@CONFDIR@/help
-	$(INSTALL) -m 0600 doc/conf/help/*.conf $(DESTDIR)@CONFDIR@/help
-	$(INSTALL) -m 0700 -d $(DESTDIR)@CONFDIR@/examples
-	$(INSTALL) -m 0600 doc/conf/examples/*.conf $(DESTDIR)@CONFDIR@/examples
 	$(INSTALL) -m 0700 unrealircd $(DESTDIR)@SCRIPTDIR@
 	$(INSTALL) -m 0700 -d $(DESTDIR)@MODULESDIR@
 	@rm -f $(DESTDIR)@MODULESDIR@/*.so 1>/dev/null 2>&1
@@ -221,30 +199,12 @@ install: all
 	$(INSTALL) -m 0700 -d $(DESTDIR)@MODULESDIR@/rpc
 	@rm -f $(DESTDIR)@MODULESDIR@/rpc/*.so 1>/dev/null 2>&1
 	$(INSTALL) -m 0700 src/modules/rpc/*.so $(DESTDIR)@MODULESDIR@/rpc
-	@#If the conf/ssl directory exists then rename it here to conf/tls
-	@#and add a symlink for backwards compatibility (so that f.e. certbot
-	@#doesn't randomly fail after an upgrade to U5).
-	-@if [ -d "$(DESTDIR)@CONFDIR@/ssl" ] ; then \
-		mv "$(DESTDIR)@CONFDIR@/ssl" "$(DESTDIR)@CONFDIR@/tls" ; \
-		ln -s "$(DESTDIR)@CONFDIR@/tls" "$(DESTDIR)@CONFDIR@/ssl" ; \
-	fi
-	$(INSTALL) -m 0700 -d $(DESTDIR)@CONFDIR@/tls
-	$(INSTALL) -m 0600 doc/conf/tls/curl-ca-bundle.crt $(DESTDIR)@CONFDIR@/tls
 	@# delete modules/cap directory, to avoid confusing with U4 to U5 upgrades:
 	rm -rf $(DESTDIR)@MODULESDIR@/cap
 	$(INSTALL) -m 0700 -d $(DESTDIR)@MODULESDIR@/third
 	@rm -f $(DESTDIR)@MODULESDIR@/third/*.so 1>/dev/null 2>&1
 	@#This step can fail with zero files, so we ignore exit status:
 	-$(INSTALL) -m 0700 src/modules/third/*.so $(DESTDIR)@MODULESDIR@/third
-	$(INSTALL) -m 0700 -d $(DESTDIR)@TMPDIR@
-	$(INSTALL) -m 0700 -d $(DESTDIR)@CACHEDIR@
-	$(INSTALL) -m 0700 -d $(DESTDIR)@PERMDATADIR@
-	$(INSTALL) -m 0700 -d $(DESTDIR)@LOGDIR@
-	-@if [ ! -f "$(DESTDIR)@CONFDIR@/tls/server.cert.pem" ] ; then \
-		$(INSTALL) -m 0600 server.req.pem $(DESTDIR)@CONFDIR@/tls ; \
-		$(INSTALL) -m 0600 server.key.pem $(DESTDIR)@CONFDIR@/tls ; \
-		$(INSTALL) -m 0600 server.cert.pem $(DESTDIR)@CONFDIR@/tls ; \
-	fi
 	@rm -f $(DESTDIR)@SCRIPTDIR@/source
 	ln -s @BUILDDIR@ $(DESTDIR)@SCRIPTDIR@/source
 	@echo ''
